src/main/java/com/zendesk/maxwell/producer/MaxwellPubsubProducer.java

```diff
@@ -0,0 +1,225 @@
+    this.succeededMessageCount = producedMessageCount;
+    this.failedMessageCount = failedMessageCount;
+    this.succeededMessageMeter = producedMessageMeter;
+    this.failedMessageMeter = failedMessageMeter;
+    this.context = context;
+  }
+
+  @Override
+  public void onSuccess(String messageId) {
+    this.succeededMessageCount.inc();
+    this.succeededMessageMeter.mark();
+
+    if ( LOGGER.isDebugEnabled() ) {
+      LOGGER.debug("->  " + this.json);
+      LOGGER.debug("    " + position);
+      LOGGER.debug("");
+    }
+
+    cc.markCompleted();
+    timer.update(cc.timeToSendMS(), TimeUnit.MILLISECONDS);
+  }
+
+  @Override
+  public void onFailure(Throwable t) {
+    this.failedMessageCount.inc();
+    this.failedMessageMeter.mark();
+
+    LOGGER.error(t.getClass().getSimpleName() + " @ " + position);
+    LOGGER.error(t.getLocalizedMessage());
+
+    if ( !this.context.getConfig().ignoreProducerError ) {
+      this.context.terminate(new RuntimeException(t));
+      return;
+    }
+
+    cc.markCompleted();
+    timer.update(cc.timeToSendMS(), TimeUnit.MILLISECONDS);
+  }
+
+}
+
+
+public class MaxwellPubsubProducer extends AbstractProducer {
+  public static final Logger LOGGER = LoggerFactory.getLogger(MaxwellPubsubProducer.class);
+
+  private final ArrayBlockingQueue<RowMap> queue;
+  private final MaxwellPubsubProducerWorker worker;
+
+  public MaxwellPubsubProducer(MaxwellContext context, Properties pubsubProperties, String pubsubProjectId, String pubsubTopic) throws IOException {
+    super(context);
+    this.queue = new ArrayBlockingQueue<>(100);
+    this.worker = new MaxwellPubsubProducerWorker(context, pubsubProperties, pubsubProjectId, pubsubTopic, this.queue);
+    this.worker.setPubsub();
+    this.worker.setDDLPubsub();
+    Thread thread = new Thread(this.worker, "maxwell-pubsub-worker");
+    thread.setDaemon(true);
+    thread.start();
+  }
+
+  @Override
+  public void push(RowMap r) throws Exception {
+    this.queue.put(r);
+  }
+
+  @Override
+  public StoppableTask getStoppableTask() {
+    return this.worker;
+  }
+}
+
+
+class MaxwellPubsubProducerWorker extends AbstractAsyncProducer implements Runnable, StoppableTask {
+  static final Logger LOGGER = LoggerFactory.getLogger(MaxwellPubsubProducer.class);
+
+  private final String projectId;
+  private Publisher pubsub;
+  private final TopicName topic;
+  private final TopicName ddlTopic;
+  private Publisher ddlPubsub;
+  private final Timer metricsTimer;
+  private final ArrayBlockingQueue<RowMap> queue;
+  private Thread thread;
+  private StoppableTaskState taskState;
+
+  public MaxwellPubsubProducerWorker(MaxwellContext context, Properties pubsubProperties, String pubsubProjectId, String pubsubTopic, ArrayBlockingQueue<RowMap> queue) {
+    super(context);
+
+    if ( pubsubTopic == null ) {
+      this.topic = TopicName.create(pubsubProjectId, "maxwell");
+    } else {
+      this.topic = TopicName.create(pubsubProjectId, pubsubTopic);
+    }
```

Comment:
`pubsubTopic`  won't be `null` if we set the default topic to `maxwell`.

Reply:
Ah! So,

1. Your comment has pointed out to me that I left the `fetchOption` lines out of this PR, just updated accordingly.

2. The Kafka producer has a default topic of `maxwell` but still checks for a `null` value, anyway--do you still want me to remove this? If so, should probably remove it from the Kafka producer, too.

Reply:
I think Kafka producer code is also having the same issue. If you can clean up here that'll be great. We can cleanup Kafka producer code later or you can clean it up if you don't mind.